packages/rpc/src/client.ts

```diff
@@ -108,173 +137,183 @@ export class RPCClient extends EventEmitter {
 		this.transport.on('message', this._onRpcMessage.bind(this));
 
 		/**
 		 * Map of nonces being expected from the transport
-		 * @type {Map}
-		 * @private
 		 */
 		this._expecting = new Map();
 
 		this._connectPromise = undefined;
-
-		/**
-		 * Map of subscriptions
-		 * @type {Map}
-		 * @private
-		 */
-		this._subscriptions = new Map();
 	}
 
 	/**
 	 * Update endpoint
+	 *
+	 * @internal
 	 */
 	public updateEndpoint(endpoint: string) {
 		this.endpoint = endpoint;
 	}
 
 	/**
 	 * Search and connect to RPC
+	 *
+	 * @param clientId - Client ID to use
 	 */
-	public connect(clientId: string) {
+	public async connect(clientId: string) {
 		if (this._connectPromise) {
 			return this._connectPromise;
 		}
+
+		const err = new Error('RPC_CONNECTION_TIMEOUT');
+		// eslint-disable-next-line @typescript-eslint/unbound-method
+		Error.captureStackTrace(err, this.connect);
+
 		this._connectPromise = new Promise((resolve, reject) => {
 			this.clientId = clientId;
-			const timeout = setTimeout(() => reject(new Error('RPC_CONNECTION_TIMEOUT')), 10e3);
+			const timeout = setTimeout(() => reject(err), 10e3);
 			timeout.unref();
 			this.once('connected', () => {
 				clearTimeout(timeout);
 				resolve(this);
 			});
 			this.transport.once('close', () => {
-				this._expecting.forEach((e) => {
-					e.reject(new Error('connection closed'));
-				});
+				for (const expect of this._expecting.values()) {
+					expect.reject(new Error('connection closed'));
+				}
+
 				this.emit('disconnected');
 				reject(new Error('connection closed'));
 			});
 			try {
 				void this.transport.connect();
-			} catch (e) {
-				reject(e);
+			} catch (error) {
+				reject(error);
 			}
 		});
 		return this._connectPromise;
 	}
 
-	/**
-	 * @typedef {RPCLoginOptions}
-	 * @param {string} clientId Client ID
-	 * @param {string} [clientSecret] Client secret
-	 * @param {string} [accessToken] Access token
-	 * @param {string} [rpcToken] RPC token
-	 * @param {string} [tokenEndpoint] Token endpoint
-	 * @param {OAuth2Scopes[]} [scopes] Scopes to authorize with
-	 */
-
 	/**
 	 * Performs authentication flow. Automatically calls Client#connect if needed.
-	 * @param {RPCLoginOptions} options Options for authentication.
+	 *
+	 * @param options - Options for authentication.
 	 * At least one property must be provided to perform login.
-	 * @example client.login({ clientId: '1234567', clientSecret: 'abcdef123' });
-	 * @returns {Promise<RPCClient>}
+	 * @example
+	 * ```js
+	 * client.login({ clientId: '1234567', clientSecret: 'abcdef123' });
+	 * ```
 	 */
 	public async login(options?: RPCLoginOptions): Promise<RPCClient> {
 		const data = await this.connect(options?.clientId ?? '');
 		if (!options?.scopes) {
 			this.emit(RPCEvents.Ready, data as MappedRPCDispatchData[RPCEvents.Ready]);
 			return this;
 		}
+
 		if (!options.accessToken) {
+			// eslint-disable-next-line require-atomic-updates
 			options.accessToken = await this.authorize(options);
 		}
```

Comment:
Not sure if the eslint-disable comment is still required
```suggestion
		options.accessToken ??= await this.authorize(options);
```